src/Twig/CookieTwigExtension.php

```diff
@@ -41,14 +49,28 @@ public function __construct(EntityManagerInterface $em, LegalCookieHelper $cooki
     public function getFunctions()
     {
         return [
-            new \Twig_SimpleFunction('get_cookie_types', [$this, 'getCookieTypes']),
-            new \Twig_SimpleFunction('get_legal_cookie', [$this, 'getLegalCookie']),
-            new \Twig_SimpleFunction('get_visitor_type', [$this, 'getVisitorType']),
-            new \Twig_SimpleFunction('legal_cookie_is_enabled', [$this, 'isLegalCookieEnabled']),
-            new \Twig_SimpleFunction('is_granted_for_cookie_bundle', [$this, 'isGrantedForCookieBundle']),
+            new TwigFunction('get_cookie_types', [$this, 'getCookieTypes']),
+            new TwigFunction('get_legal_cookie', [$this, 'getLegalCookie']),
+            new TwigFunction('get_visitor_type', [$this, 'getVisitorType']),
+            new TwigFunction('cookie_in_domain', [$this, 'cookieAllowedForDomain']),
+            new TwigFunction('legal_cookie_is_enabled', [$this, 'isLegalCookieEnabled']),
+            new TwigFunction('is_granted_for_cookie_bundle', [$this, 'isGrantedForCookieBundle']),
         ];
     }
 
+    public function cookieAllowedForDomain(Cookie $cookie)
+    {
+        if($this->domainConfiguration->isMultiDomainHost() === false) {
+            return true;
+        }
+
+        $host = $this->domainConfiguration->getHost();
+        $config = $this->domainConfiguration->getFullHostConfig();
+        $currentDomain = $config[$host]['id'];
+
+        return empty($cookie->getDomain()) || $cookie->getDomain() === $currentDomain;
```

Comment:
split this check into an early return before the domain configuration calls. Because if the cookie is not assign to a specific domain it is always true

```php
if (empty($cookie->getDomain())) {
     return true;
}
```